File: uiComponents/src/main/java/com/yuriisurzhykov/pointsdetector/uicomponents/list/MutableListAdapter.java
package com.yuriisurzhykov.pointsdetector.uicomponents.list;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.List;

public interface MutableListAdapter<T> {

    void submitList(@Nullable List<T> list);

    void removeItem(@NonNull T item);

    void removeItem(int position);

    void insertItem(@NonNull T item, int position);

    @NonNull
    List<T> getCurrentList();

}

File: ui-components/src/main/java/com/yuriisurzhykov/pointdetector/uicomponents/list/MutableListAdapter.kt
package com.yuriisurzhykov.pointdetector.uicomponents.list

interface MutableListAdapter<T> {
    fun submitList(list: List<T>?)
    fun removeItem(item: T)
    fun removeItem(position: Int)
    fun insertItem(item: T, position: Int)
    fun getCurrentList(): List<T>
}